osquery/tables/system/windows/prefetch.cpp

```diff
@@ -0,0 +1,370 @@
+    DWORD version) {
+  PrefetchVolumeInfo result;
+
+  const auto volume_header_size = (version == 30) ? 104 : 96;
+  const auto volume_offset = prefetch_file_info->VolumeInformationOffset;
+  // Size is given in bytes.
+  const auto volume_size = prefetch_file_info->VolumesInformationSize;
+  if (volume_offset > data.size()) {
+    // Unexpected offset.
+    return result;
+  }
+
+  // Headers are stacked sequentially.
+  auto volume_header_offset = volume_offset;
+
+  std::vector<std::string> directories;
+  std::vector<std::string> volume_serials;
+  std::vector<std::string> volume_creations;
+  for (size_t i = 0; i < prefetch_file_info->NumberOfVolumes; i++) {
+    if (volume_header_offset + volume_header_size > data.size()) {
+      // Unexpected size.
+      return result;
+    }
+
+    const auto prefetch_volume_info =
+        (PPREFETCH_VOLUME_INFORMATION)(&data[0] + volume_header_offset);
+    const auto creation =
+        filetimeToUnixtime(prefetch_volume_info->VolumeCreationTime);
+    volume_creations.push_back(std::to_string(creation));
+    const auto serial = prefetch_volume_info->VolumeSerialNumber;
+    volume_serials.push_back((boost::format("%x") % serial).str());
+    volume_header_offset += volume_header_size;
+
+    const auto dir_count = prefetch_volume_info->NumberOfDirectoryStrings;
+    size_t dir_offset = prefetch_volume_info->DirectoryStringsOffset;
+    if (dir_count == 0) {
+      continue;
+    }
+
+    for (size_t j = 0; j < dir_count; j++) {
+      if (volume_offset + dir_offset + sizeof(PDIRECTORY_STRING) >
+          data.size()) {
+        // Unexpected offset.
+        break;
+      }
+
+      const auto prefetch_directory =
+          (PDIRECTORY_STRING)(&data[0] + volume_offset + dir_offset);
+      dir_offset += sizeof(DIRECTORY_STRING);
+
+      auto length = wcsnlen_s(prefetch_directory->Directory,
+                              (volume_size - dir_offset) / 2);
+      if (length == 0 || length == (volume_size - dir_offset) / 2) {
+        // A null wide character was not found.
+        break;
+      }
+
+      auto filename = wstringToString(prefetch_directory->Directory);
+      directories.emplace_back(std::move(filename));
+      dir_offset += (length + 1) * 2;
+    }
+  }
+
+  result.volume_creation = osquery::join(volume_creations, ",");
+  result.volume_serial = osquery::join(volume_serials, ",");
+  result.directories = osquery::join(directories, ",");
+  result.directories_count = directories.size();
+  return result;
+}
+
+void parsePrefetchData(RowYield& yield,
+                       const std::vector<UCHAR>& data,
+                       const std::string& file_path) {
+  const auto prefetch_header = (PPREFETCH_FILE_HEADER)&data[0];
+  if (prefetch_header->Signature != PREFETCH_SIGNATURE) {
+    LOG(INFO) << "Unsupported prefetch file header: " << file_path;
+    return;
+  }
+
+  const auto version = prefetch_header->Version;
+  if (version != 30 && version != 23 && version != 26) {
```

Comment:
let's define these on top,
e.g.
```
const unsigned long kPrefetchVersionWindows10 =  30;
...
```

and then use if `(version != kPrefetchVersionWindows10 ...)`

Reply:
good idea, added constants